### PR TITLE
Document how to customize the layout

### PR DESCRIPTION
**Pull request by jide**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -207,6 +207,23 @@ module.exports = {
 }
 ```
 
+This can also be used to customize the layout by duplicating the original [layout.js](https://raw.githubusercontent.com/kadirahq/react-storybook/master/src/client/ui/layout.js) file at `.storybook/layout.js` and setting webpack config like this :
```
Comment (arunoda) [inline docs/api.md]: Could you add a small title on top of this. Like: Customizing The UI.

**Comment by jide** [comment]
I added a title and rephrased a bit.

**Comment by arunoda** [comment]
Awesome.

**Comment by clauderic** [comment]
Can this be updated? It the link to [layout.js](https://raw.githubusercontent.com/kadirahq/react-storybook/master/src/client/ui/layout.js) is broken

**Comment by jide** [comment]
@clauderic The documentation for this has moved to https://github.com/kadirahq/react-storybook/blob/master/docs/configure_storybook.md#customizing-the-ui

**Comment by clauderic** [comment]
Great! That still does not help though, the link is still broken over there 😿

**Comment by jide** [comment]
Oh I see, yeah @arunda made significant changes recently. I guess it's the https://github.com/kadirahq/react-storybook/blob/master/src/client/manager/modules/ui/components/layout/index.js file now, but I did not dig more - you should be able to figure out the right configuration by changing the paths in the config and using this file instead.

**Comment by clauderic** [comment]
Hmmm okay. Doesn't seem to work anymore. It'd be great if the documentation on this was revised

**Comment by uipoet** [comment]
Attempted replacing a UI module following documentation and nothing happens, without errors.

**Comment by joaomilho** [comment]
+1

**Comment by csalvato** [comment]
Where's the documentation for this now? All links are dead. :)

**Comment by jeroenransijn** [comment]
I would like to know what the progress is on this?

**Comment by ndelangen** [comment]
@jeroenransijn Could you open a new issue on this describing what's wrong? I'm not sure what to do with this now, and It's hard to keep track of a merged PR. Thanks!

**Comment by binomialstew** [comment]
I know the last question here is from over a year ago, but for anyone finding this issue that would like to use their own custom layout, you can use the `webpack.NormalModuleReplacementPlugin` to replace the import of `layout.js` via @storybook. For example, in your webpack config:
```js
import { NormalModuleReplacementPlugin } from webpack
...
module.exports = (storybookBaseConfig, env) => {
  storybookBaseConfig.plugins = [
    // replace the import (in routes @storybook/ui/dist/modules/ui/routes.js) with your custom layout alias
    new NormalModuleReplacementPlugin(/^\.\/containers\/layout$/, 'custom-layout')
  ]
  ...
  storybookBaseConfig.resolve = {
    alias: {
      'custom-layout': path.resolve(project.basePath, '.storybook/layout.js')
    }
  }
}
```

**Comment by ndelangen** [comment]
Working on refactoring and improving the UI right now,

Please share your ideas and customisations so I know what users want.

See:
https://github.com/storybooks/storybook/pull/4086#issuecomment-420931454

**Comment by binomialstew** [comment]
I would like the ability to disable the mobile layout set by `isMobileDevice`. I developed a custom mobile layout before it was available in storybook. The new release with the mobile layout overrides my `addon-options` config and breaks my layout.

**Comment by binomialstew** [comment]
@ndelangen, the issue I have is more the implementation of the mobile layout rather than the UI itself. My main point of issue is it's triggered by user agent rather than css media queries. In my view, the default layout works well for iPad—the screen width is sufficient for the left sidebar, but still, the mobile layout is triggered and shown on iPads.

Another issue I had—the mobile layout seems to override any options that were set initially, like "showAddonPanel" or "goFullScreen". I had previously set up a button to toggle the addons panel, and this negates that. 

I set up my storybook with media queries that hide the left sidebar and show a simplified header with a menu button where the stories are accessible—but only when the viewport is reduced to a specific width.

In general, I think that determining responsive layouts with CSS rather than user agent is a better approach because what's important is usually the screen size, more than anything else. For me, having an option to disable `isMobileDevice` would be sufficient, but if this were to be redone, I think the best approach would be to handle responsivity with media queries and also add an option to disable mobile layout.

**Comment by ndelangen** [comment]
@binomialstew the media-query issue is addressed in my UI refactor.

the new mobile layout in this overhaul will show the component by default when loaded, regardless of settings, I might change that.

**Comment by ndelangen** [comment]
#4086

**Pull request merged**